### PR TITLE
Investigate 1954 certificate always regenerated

### PR DESCRIPTION
**Pull request by dgageot**
So, this does not solve anything but adds more logs during the certificates validation.

### CODE REVIEW

File: commands/commands.go [hunk 1]
```diff
@@ -419,6 +419,7 @@ func runActionForeachMachine(actionName string, machines []*host.Host) []error {
 		go machineCommand(actionName, machine, serialChan)
 		if err := <-serialChan; err != nil {
 			log.Errorln(err)
+			errs = append(errs, err)
```
Comment (nathanleclaire) [inline commands/commands.go]: Ah, nice catch

File: libmachine/cert/cert.go [hunk 2]
```diff
@@ -195,12 +197,16 @@ func ValidateCertificate(addr, caCertPath, serverCertPath, serverKeyPath string)
 		return false, ErrValidatingCert{err}
 	}
 
+	log.Debugf("Configuration: +%v", tlsConfig)
+
 	dialer := &net.Dialer{
-		Timeout: time.Second * 2,
+		Timeout: time.Second * 5,
+		DualStack: true,
```
Comment (nathanleclaire) [inline libmachine/cert/cert.go]: So what's the issue which was going wrong that this solves?

Comment (dgageot) [inline libmachine/cert/cert.go]: @nathanleclaire none ;-) I was just trying random things with the dialer but it didn't help

File: libmachine/cert/cert.go [hunk 3]
```diff
@@ -195,12 +207,15 @@ func ValidateCertificate(addr, caCertPath, serverCertPath, serverKeyPath string)
...
 
+	log.Debugf("Configuration: +%v", tlsConfig)
+
 	dialer := &net.Dialer{
 		Timeout: time.Second * 2,
 	}
 
 	_, err = tls.DialWithDialer(dialer, "tcp", addr, tlsConfig)
 	if err != nil {
+		log.Debugf("Certs are not valid: %s", err)
```
Comment (blaggacao) [inline libmachine/cert/cert.go]: @carolynvs: Does your output refer to this line: https://github.com/docker/machine/issues/1954#issuecomment-150039974 ?

Comment (carolynvs) [inline libmachine/cert/cert.go]: Yes line 216 from this diff above causes that message to be printed on the client and on the server I see that error about cert dates in docker.log on the host.

Comment (blaggacao) [inline libmachine/cert/cert.go]: @dgageot @nathanleclaire We should probably keep this debugging line in the release? It helps debugging when people are not capable of building some debugging PR by posting those lines...

Comment (nathanleclaire) [inline libmachine/cert/cert.go]: Sure, more debugging output usually helps.

**Comment by dgageot** [comment]
@nathanleclaire @dmp42 Can you please take a look?

**Comment by dmp42** [comment]
LGTM
Thanks @dgageot !

**Comment by blaggacao** [comment]
we are hunting this down, it seems a valid date issue, boot2docker having a bad `date` #1954

**Comment by nathanleclaire** [comment]
too late but LGTM

**Pull request merged**